OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ApplicationController.cs

```diff
@@ -393,12 +393,12 @@ public async Task<IActionResult> Update(ApplicationUpdate applicationDto)
             var result =
                 await applicationService.Update(applicationDto, workshop.ProviderId).ConfigureAwait(false);
 
-            if (!result.Succeeded)
+            return result.Match<ActionResult>(
+            error => StatusCode((int)error.HttpStatusCode, new { error.Message, error.ApiErrorResponse }),
+            result =>
             {
-                return BadRequest(result.OperationResult.Errors.ElementAt(0).Description);
-            }
-
-            return Ok(result.Value);
+                return Ok(result.Value);
+            });
```

Comment:
(Excessive code) Please use expression-bodied methods when it's one short line of implementation, as it is here, thus you could omit curly braces and just write `result => Ok(result.Value);`

Reply:
Done